tensorrt_llm/_torch/models/modeling_gemma4mm.py

```
@@ -550,6 +545,245 @@ def call_with_text_prompt(
+            for multimodal_param in multimodal_params
+        ]
+
+        # The cache helper splits this tensor by request. Preserve that order while still batching
+        # consecutive requests with the same modality.
+        if (
+            all(len(modalities) == 1 for modalities in param_modalities)
+            and len({modalities[0] for modalities in param_modalities}) > 1
+        ):
+            embeddings = []
+            for _, param_group in groupby(
+                zip(param_modalities, multimodal_params), key=lambda item: item[0][0]
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)